handler/src/main/java/io/netty/handler/ssl/ReleasableOpenSslServerContext.java

```diff
@@ -0,0 +1,186 @@
+            long sessionCacheSize, long sessionTimeout, ClientAuth clientAuth) throws SSLException {
+        super(ciphers, cipherFilter, apn, sessionCacheSize, sessionTimeout, SSL.SSL_MODE_SERVER, keyCertChain,
+              clientAuth, true);
+        // Create a new SSL_CTX and configure it.
+        boolean success = false;
+        try {
+            ServerContext context = newSessionContext(this, ctx, engineMap, trustCertCollection, trustManagerFactory,
+                                                      keyCertChain, key, keyPassword, keyManagerFactory);
+            sessionContext = context.sessionContext;
+            keyMaterialManager = context.keyMaterialManager;
+            success = true;
+        } finally {
+            if (!success) {
+                release();
+            }
+        }
+    }
+
+    @Override
+    public OpenSslServerSessionContext sessionContext() {
+        return sessionContext;
+    }
+
+    @Override
+    OpenSslKeyMaterialManager keyMaterialManager() {
+        return keyMaterialManager;
+    }
+
+    static final class ServerContext {
+        OpenSslServerSessionContext sessionContext;
+        OpenSslKeyMaterialManager keyMaterialManager;
+    }
+
+    static ServerContext newSessionContext(ReleasableOpenSslContext thiz, long ctx, OpenSslEngineMap engineMap,
+                                           X509Certificate[] trustCertCollection,
+                                           TrustManagerFactory trustManagerFactory,
+                                           X509Certificate[] keyCertChain, PrivateKey key,
+                                           String keyPassword, KeyManagerFactory keyManagerFactory)
+            throws SSLException {
+        ServerContext result = new ServerContext();
+        synchronized (ReleasableOpenSslContext.class) {
+            try {
+                SSLContext.setVerify(ctx, SSL.SSL_CVERIFY_NONE, VERIFY_DEPTH);
+                if (!OpenSsl.supportsKeyManagerFactory()) {
+                    if (keyManagerFactory != null) {
+                        throw new IllegalArgumentException(
+                                "KeyManagerFactory not supported");
+                    }
+                    checkNotNull(keyCertChain, "keyCertChain");
+
+                        /* Set certificate verification policy. */
+                    SSLContext.setVerify(ctx, SSL.SSL_CVERIFY_NONE, VERIFY_DEPTH);
+
+                    setKeyMaterial(ctx, keyCertChain, key, keyPassword);
+                } else {
+                    // javadocs state that keyManagerFactory has precedent over keyCertChain
+                    if (keyManagerFactory == null && keyCertChain != null) {
+                        keyManagerFactory = buildKeyManagerFactory(
+                                keyCertChain, key, keyPassword, keyManagerFactory);
+                    }
+                    X509KeyManager keyManager = chooseX509KeyManager(keyManagerFactory.getKeyManagers());
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'keyManagerFactory' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2259)